koku/masu/processor/ocp/ocp_report_db_cleaner.py

```diff
@@ -147,3 +148,25 @@ def purge_expired_report_data_by_date(self, expired_date, simulate=False):
                 LOG.info(log_json(msg="deleted table partitions", count=del_count, schema=self._schema))
 
         return removed_items
+
+    def purge_expired_trino_partitions(self, expired_date, provider_uuid=None, simulate=False):
+        """Removes expired trino partitions."""
+        LOG.debug(f"purge_expired_trino_partitions: {expired_date}, {provider_uuid}, {simulate}")
+        if expired_date is None and provider_uuid is None:
```

Comment:
If one of `expired_date` or `provider_uuid` is required, then `expired_date` cannot be a required parameter. I'm also assuming it's ok to provide both values and that `None` is not valid.
```suggestion
    def purge_expired_trino_partitions(self, expired_date=None, provider_uuid=None, simulate=False):
        """Removes expired trino partitions."""
        LOG.debug(f"purge_expired_trino_partitions: {expired_date}, {provider_uuid}, {simulate}")
        require_one_of = (expired_date, provider_uuid)
        if all(value is None for value in require_one_of):
```

If you want to check if a parameter was provided in addition to its value, you can use a sentinel object instead of `None` as the default value.

Reply:
Where is `provider_uuid` used?

Reply:
Good call out Sam, I went ahead and removed that param. 🙇